### PR TITLE
[arc-multicloud] Improvement, Bugfix and Add Customization

### PR DESCRIPTION
**Pull request by alexmahonic**
### Changes introduced:
* GA release
* Register all required RPs before running each command
* Reflect API swagger changes
* Store the output of ``generate-aws-template`` to json file on disk
* Update help guide with more examples

### Azure Arc Team reviewer instructions
Code changes under ``src/multicloud-connector/azext_multicloud_connector/aaz`` are auto generated, so correctness review for those files is not necessary. However, you may also review on the wording/description of the help and examples

#### Register the RPs
5d8bc138-1abd-43c4-9bab-815c50a61002
347ecac6-87ef-4c28-b317-a40d373cea13

#### generate-aws-template & Store AWS template to local disk
image
image

image
image

#### PCC create
image

#### PCC test-permission
image

#### SC sync-now
image

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
arc-multicloud

File: src/multicloud-connector/HISTORY.rst
Comment (AllyW) [inline src/multicloud-connector/HISTORY.rst:8]: 1.0.0b1 -> GA version:  1.0.0
Please remove `azext.isPreview: true` tag in  azext_metadata.json file as well. 
For more detail, please refer to doc here: https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md

Comment (alexmahonic) [reply src/multicloud-connector/HISTORY.rst:8]: Updated the release version to 1.0.0

File: src/multicloud-connector/azext_multicloud_connector/_utils.py
Comment (AllyW) [inline src/multicloud-connector/azext_multicloud_connector/_utils.py:78]: ValidationError only accepts one error_msg

File: src/multicloud-connector/azext_multicloud_connector/commands.py
Comment (AllyW) [inline src/multicloud-connector/azext_multicloud_connector/commands.py:15]: For new added functionality, please add test for this cmd.

File: src/multicloud-connector/azext_multicloud_connector/aaz/latest/arc_multicloud/_generate_aws_template.py
Comment (AllyW) [inline src/multicloud-connector/azext_multicloud_connector/aaz/latest/arc_multicloud/_generate_aws_template.py:55]: The deletion of this arg is by design?

File: src/multicloud-connector/HISTORY.rst
Comment (AllyW) [inline src/multicloud-connector/HISTORY.rst:12]: Since the aaz autogenerated cmd's examples are changed, please make a pr to aaz repo to update the related aaz models here: https://github.com/Azure/aaz/tree/main/Commands/arc-multicloud.

If the model is already updated, please link the aaz repo update pr here.

Comment (alexmahonic) [reply src/multicloud-connector/HISTORY.rst:12]: FYI - https://github.com/Azure/aaz/pull/577

File: src/multicloud-connector/azext_multicloud_connector/_utils.py
Comment (alexmahonic) [reply src/multicloud-connector/azext_multicloud_connector/_utils.py]: Updated

**Review by AllyW — APPROVED** [review]

File: src/multicloud-connector/azext_multicloud_connector/aaz/latest/arc_multicloud/solution_configuration/_create.py
Comment (bagajjal) [inline src/multicloud-connector/azext_multicloud_connector/aaz/latest/arc_multicloud/solution_configuration/_create.py:21]: Try to capture all the available options 

image

Comment (bagajjal) [reply src/multicloud-connector/azext_multicloud_connector/aaz/latest/arc_multicloud/solution_configuration/_create.py:21]: Also create a work item to link the public documentation that lists all the available options with default values. Please work with Meagan on this.

Comment (bagajjal) [inline src/multicloud-connector/azext_multicloud_connector/aaz/latest/arc_multicloud/solution_configuration/_create.py:21]: Bug. solutionconfiguration creation fails when tried with example given in the help message.

image

File: src/multicloud-connector/azext_multicloud_connector/_utils.py
Comment (bagajjal) [inline src/multicloud-connector/azext_multicloud_connector/_utils.py:25]: when is this called? can you add few comments.

File: src/multicloud-connector/azext_multicloud_connector/aaz/latest/arc_multicloud/public_cloud_connector/_delete.py
Comment (bagajjal) [inline src/multicloud-connector/azext_multicloud_connector/aaz/latest/arc_multicloud/public_cloud_connector/_delete.py:22]: rgpublicCloud -> multiCloudRG (Lets use multicloud everywhere instead of publicCloud)
myConnector -> awsConnector

Comment (bagajjal) [reply src/multicloud-connector/azext_multicloud_connector/aaz/latest/arc_multicloud/public_cloud_connector/_delete.py:22]: Please take this as global comment.

**Comment by AllyW** [comment]
hi @bagajjal and @alexmahonic , merged this pr already since all tests passed and no more comments this morning when I review it. If more changes expected, please raise a new pr to update it. Thanks. 

If any other concerns, please let me know.

**Comment by bagajjal** [comment]
@AllyW , one suggestion is to wait for the approval from product team members before merging the PR.
I don't know how much you cover as part of your review.
Product team review adds lot of value and provides true feedback.

**Comment by alexmahonic** [comment]
> hi @bagajjal and @alexmahonic , merged this pr already since all tests passed and no more comments this morning when I review it. If more changes expected, please raise a new pr to update it. Thanks.
> 
> If any other concerns, please let me know.

I'll check those comments and update in following PRs if needed. It shouldn't have product impact at the moment.

**Comment by alexmahonic** [comment]
Synced with @bagajjal with the comments and will address and investigate accordingly.

**Pull request merged**
